toqito/matrix_ops/tensor_unravel.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+
+def tensor_unravel(constraint_tensor: np.ndarray) -> np.ndarray:
+    r"""Decode a clause tensor (indicator tensor) into its raw 1D representation.
+
+    In binary constraint system (BCS) games, parity constraints can be encoded as
+    **clause tensors** — n-dimensional NumPy arrays of shape `(2, 2, ..., 2)`,
+    filled with a constant background value (e.g., `(-1)**b[i]`) except for a
+    single unique entry that marks the satisfying assignment.
+
+    This function unravels such a tensor by:
+       1. Locating the unique element (the one appearing exactly once).
+       2. Extracting its multi-dimensional index `(i1, i2, ..., in)`.
+       3. Returning a 1D NumPy array `[i1, i2, ..., in, value]`, where the first `n`
+         entries are the coordinates and the last entry is the unique value (±1).
+
+    Conceptually, this is a form of structured tensor decoding, closely related to:
+       - Indicator (Kronecker delta) tensors in multilinear algebra (see:cite:`Kolda_2009_Tensor`)
+       - The matrix `vec`-operator for flattening matrices (see:cite:`Horn_1985_Matrix`)
+       - Parity-projector encodings in linear-system games (see:cite:`Cleve_2016_Perfect`)
+    
+    Examples
+    ==========
+    .. jupyter-execute::
+    
+     import numpy as np
+     from toqito.matrix_ops.tensor_unravel import tensor_unravel
+          
+     tensor_constraint = np.array([[-1, -1], [-1, 1]])
+     tensor_unravel(tensor_constraint)
+        
+     The tensor-form constraint representation is commonly used in implementations of
+     binary constraint system (BCS) games. For background on BCS games, see:cite:`Cleve_2014_Characterization`.
```

Comment:
RTD is still failing: 
https://app.readthedocs.org/api/v2/build/28705699.txt

```suggestion
    .. jupyter-execute::
        import numpy as np
        from toqito.matrix_ops.tensor_unravel import tensor_unravel
          
        tensor_constraint = np.array([[-1, -1], [-1, 1]])
        tensor_unravel(tensor_constraint)
        
    The tensor-form constraint representation is commonly used in implementations of
    binary constraint system (BCS) games. For background on BCS games, see:cite:`Cleve_2014_Characterization`.
```